netcfgbu/connectors/basic.py

```diff
@@ -333,38 +324,49 @@ async def login(self):
             reason=f"No valid username/password, attempted {len(self.creds)} credentials."
         )
 
-    async def close(self):
+    async def close(self) -> None:
+        """Closes the SSH connection."""
         self.conn.close()
         await self.conn.wait_closed()
-        self.log.info(f"CLOSED: {self.name}")
+        self.log.info("CLOSED: %s", self.name)
 
     # -------------------------------------------------------------------------
     #
     #                                    Helpers
     #
     # -------------------------------------------------------------------------
 
-    async def read_until_prompt(self):
+    async def read_until_prompt(self) -> bytes:
+        """Reads from the SSH process until the prompt is detected.
+
+        Returns:
+            The output read until the prompt is found as a byte string.
+        """
         output = b""
         while True:
+            self.log.debug("%s - %s", self.name, output)
             output += await self.process.stdout.read(io.DEFAULT_BUFFER_SIZE)
             nl_at = output.rfind(b"\n")
-            if mobj := self.PROMPT_PATTERN.match(output[nl_at + 1 :]):
+            if mobj := self.prompt_pattern.match(output[nl_at + 1 :]):
                 self._cur_prompt = mobj.group(1)
                 return output[0:nl_at]
 
-    async def run_command(self, command):
+    async def run_command(self, command: str) -> bytes:
+        """Runs a command on the SSH process and reads the output until the prompt is detected.
+
+        Args:
+            command: The command to run on the SSH device.
+
+        Returns:
+            The output of the command as a byte string.
+        """
         wr_cmd = command + "\n"
         self.process.stdin.write(wr_cmd.encode("utf-8"))
         output = await self.read_until_prompt()
         return output[len(wr_cmd) + 1 :]
 
-    async def run_disable_paging(self):
-        """
-        This coroutine is used to execute each of the `pre_get_config` commands
-        so that the CLI will not prompt for "--More--" output.
-        """
-
+    async def run_disable_paging(self) -> None:
+        """Executes the pre-get configuration commands to disable paging on the device."""
         disable_paging_commands = self.os_spec.pre_get_config
         if not isinstance(disable_paging_commands, list):
             disable_paging_commands = [disable_paging_commands]
```

Comment:
**Suggestion:** In the `run_disable_paging` method, consider logging the commands being run for better traceability and debugging. [enhancement, importance: 5]
```suggestion
disable_paging_commands = self.os_spec.pre_get_config
self.log.debug(f"Running disable paging commands: {disable_paging_commands}")
```